src/query.rs

```diff
@@ -240,14 +240,24 @@ mod tests {
             ron::from_str(&expected_result_text)
                 .expect("could not parse expected outputs as ron format");
 
-        let adapter = VersionedRustdocAdapter::new(&current, Some(&baseline))
-            .expect("could not create adapter");
-        let results_iter = adapter
-            .run_query(&semver_query.query, semver_query.arguments.clone())
-            .unwrap();
-
-        let mut actual_results: Vec<BTreeMap<_, _>> = results_iter
-            .map(|res| res.into_iter().map(|(k, v)| (k.to_string(), v)).collect())
+        let mut actual_results: Vec<BTreeMap<_, _>> = get_test_crate_names()
+            .into_iter()
+            .flat_map(|crate_pair| {
```

Comment:
A better idea might be to not flatten it, but leave the information about which crate those outputs are.